### DISCUSSION TITLE
Center cmdline popup

### DISCUSSION THREAD

**Discussion by jeeeem** [folke/noice.nvim#234] [Ideas]
How to center the cmdline when `command_palette` is set to `true`
![image](https://user-images.githubusercontent.com/51053482/202844348-1a6456a2-fdc8-4342-9809-d1a1cece40b3.png)

**Comment by folke** [comment]
What do you mean? It is centered in your screenshot? Or do you mean `config.views.cmdline_popup.align = "center"`?

**Reply by jeeeem** [reply]
> What do you mean? It is centered in your screenshot? Or do you mean `config.views.cmdline_popup.align = "center"`?

I mean I want it to be centered vertically and horizontally. 

better screenshot 
*I wanted it to be in the red line*
![image](https://user-images.githubusercontent.com/51053482/202908901-1f61e89c-3fd0-4936-b995-676a2ff48836.png)

`config.views.cmdline_popup.align = "center"` tried this one but it doesn't work

**Reply by jeeeem** [reply]
I just read again the README and the code, I realized I can create my own presets.

I just copy the default preset and change the `position.row` for cmdline_popup and popupmenu.
```
  presets = {
    bottom_search = false, -- use a classic bottom cmdline for search
    -- command_palette = true, -- position the cmdline and popupmenu together
    long_message_to_split = true, -- long messages will be sent to a split
    inc_rename = false, -- enables an input dialog for inc-rename.nvim
    lsp_doc_border = true, -- add a border to hover docs and signature help
    command_palette = {
      views = {
        cmdline_popup = {
          position = {
            row = 20,
            col = "50%",
          },
          size = {
            min_width = 60,
            width = "auto",
            height = "auto",
          },
        },
        popupmenu = {
          relative = "editor",
          position = {
            row = 23,
            col = "50%",
          },
          size = {
            width = 60,
            height = "auto",
            max_height = 15,
          },
          border = {
            style = "rounded",
            padding = { 0, 1 },
          },
          win_options = {
            winhighlight = { Normal = "Normal", FloatBorder = "NoiceCmdlinePopupBorder" },
          },
        },
      },
    },
  },
```
It looked like this now
![image](https://user-images.githubusercontent.com/51053482/202910425-e335632d-fb9e-49aa-8708-b6590b51e9c9.png)

**Reply by JuanZoran** [reply]
Thanks for that!  I am confused with the same problem........

**Reply by chaitanyabsprip** [reply]
how do you have that kind of completion? I am using cmp and it's the plain old cmp completion

**Reply by folke** [reply]
@Chaitanyabsprip that's the default Noice completion. Disable `cmp-cmdline` to get it

**Reply by chaitanyabsprip** [reply]
Currently, it's triggered on Tab, can I configure it to be triggered automatically as I type?

**Reply by folke** [reply]
the Noice popup **IS** the native cmdline popup, so any config for that apply for Noice. As far as I know, Neovim does not support showing the pmenu when typing. You can change the keybinding however.

**Reply by chaitanyabsprip** [reply]
any way I can translate noice.nvim's ui to cmp?

**Reply by folke** [reply]
no

**Reply by chaitanyabsprip** [reply]
I tried the configuration that jeeeem shared but that isn't working for me

<img width="1440" alt="Screenshot 2023-01-08 at 03 00 17" src="https://user-images.githubusercontent.com/55058351/211171025-e2013e97-f682-4d17-8ae1-a76abc64882f.png">
